### PR TITLE
Edit Product: Enable search categories

### LINKED ISSUE #11379: Edit Product: Add search bar to category list
User review:

> Hello, It's very easy to get the whole data of our store at this app. But I'm facing one issue which is search category option while selecting specific category among many I need to scroll down and see manually rather than this I want search icon so that I can easy type and find relevant category and it will make my work faster .... I hope team will work on it and make this option available in the next update Thank you

### PR DESCRIPTION
**Pull request by itsmeichigo**
Closes: #11379 

## Description

This PR enables search categories for editing products. There's an existing bool check for this so the fix is simple. I updated the UI to move the Add Category button to the bottom of the screen instead of on the top so that the screen looks nicer with the search bar on the top.

Also enabled the Clear selection button and fixed the missing leading margin for the button.

## Testing instructions

- Log in to a store and switch to the Products tab.
- Select an existing product or create a new one.
- On the product form, select Categories (or go to Add more details > Categories).
- The edit category screen should be updated with the search bar on the top and the Add button at the bottom.
- Select any category, the Clear selection button should show up. 
- Tap the Clear selection, all categories should be unselected.

## Screenshots

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: RELEASE-NOTES.txt [hunk 1]
```diff
@@ -9,6 +9,7 @@
 - [*] Products: Downloadable products now accept file types other than images from WordPress media library. [https://github.com/woocommerce/woocommerce-ios/pull/11356]
 - [*] Payments menu: restored the ability to search for Payments in device Spotlight. [https://github.com/woocommerce/woocommerce-ios/pull/11343]
 - [*] Payments menu: show the selected payment gateway when there's more than one to choose from [https://github.com/woocommerce/woocommerce-ios/pull/11345]
+- [*] Edit Products: category list is now searchable. [https://github.com/woocommerce/woocommerce-ios/pull/11380]
```
Comment (selanthiraiyan) [inline RELEASE-NOTES.txt]: 🔔 We can move this to 16.7

**Review by selanthiraiyan — APPROVED** [review]
Nice addition! 🎉  

LGTM :shipit: 

---

Just curious, do you have context about why search wasn't enabled (using `searchEnabled` parameter) before? Was it due to any missing UI work?

File: RELEASE-NOTES.txt (hunk 1 above)
Comment (itsmeichigo) [reply RELEASE-NOTES.txt]: Moved in 51bbf2ff429c5f67a918190f25f0c55a9915efc0 ✅

**Comment by itsmeichigo** [comment]
> Just curious, do you have context about why search wasn't enabled (using searchEnabled parameter) before? Was it due to any missing UI work?

Good question! From gitblame, looks like I [added](https://github.com/woocommerce/woocommerce-ios/pull/6785) the search bar as part of the Coupons project and left the UI of the product category list as-is.

**Pull request merged**